Plugins/BuildServerIntegration/AzureDevOpsIntegration/AzureDevOpsAdapter.cs

```diff
@@ -158,6 +229,22 @@ private async Task ObserveBuildsAsync(DateTime? sinceDate, bool running, IObserv
             }
         }
 
+        private void BtnIgnoreError_Click(object sender, EventArgs e)
+        {
+            CloseTaskDialogWithResult(sender, TaskDialogResult.No);
+        }
+
+        private void BtnUpdateToken_Click(object sender, EventArgs e)
+        {
+            CloseTaskDialogWithResult(sender, TaskDialogResult.Yes);
+        }
+
+        private static void CloseTaskDialogWithResult(object sender, TaskDialogResult taskDialogResult)
+        {
+            var taskDialog = (TaskDialog)((TaskDialogButton)sender).HostingDialog;
+            taskDialog.Close(taskDialogResult);
+        }
```

Comment:
Please use local functions and capture locals, the same way as used in other places. This will make it simpler to migrate to the Windows Forms Task Dialog.